eq-author/src/App/QuestionnaireDesignPage/index.test.js

```diff
@@ -160,6 +161,78 @@ describe("QuestionnaireDesignPage", () => {
     });
   });
 
+  describe("onAddCalculatedSummaryPage", () => {
+    it("should add new page below current page", () => {
+      wrapper.find(NavigationSidebar).simulate("addCalculatedSummaryPage");
+
+      expect(mockHandlers.onAddCalculatedSummaryPage).toHaveBeenCalledWith(
+        section.id,
+        page.position + 1
+      );
+    });
+
+    it("should add page at start of section if page not found", () => {
+      wrapper.setProps({
+        questionnaire: {
+          ...questionnaire,
+          sections: [
+            {
+              ...section,
+              pages: [],
+            },
+          ],
+        },
+      });
+
+      wrapper.find(NavigationSidebar).simulate("addQuestionPage");
+
+      wrapper.find(NavigationSidebar).simulate("addCalculatedSummaryPage");
+
+      expect(mockHandlers.onAddCalculatedSummaryPage).toHaveBeenCalledWith(
+        section.id,
+        1
+      );
+    });
+
+    it("should be able to add a page at the start when on a section", () => {
+      wrapper.setProps({
+        match: {
+          params: {
+            questionnaireId: questionnaire.id,
+            entityName: SECTION,
+            entityId: section.id,
+          },
+        },
+      });
+
+      wrapper.find(NavigationSidebar).simulate("addCalculatedSummaryPage");
+
+      expect(mockHandlers.onAddCalculatedSummaryPage).toHaveBeenCalledWith(
+        section.id,
+        0
+      );
+    });
+
+    it("should be able to add a page after the confirmation when on a confirmation page", () => {
```

Comment:
Confirmation doesn't ever have any effect on this in author. This test IMO is not worthwhile.